apps/etl-func/src/adapters/functions/messages-ingestion.ts

```diff
@@ -11,16 +11,24 @@ const logger = pino({
   level: process.env.NODE_ENV === "production" ? "error" : "debug",
 });
 
+const processMessageMetadata = (
+  input: unknown,
+): MessageMetadata | undefined => {
+  const result = messageMetadataSchema.safeParse(input);
+  //we are considering, and returning, only processed messages. A processed message has isPending === false
+  if (result.success && !result.data.isPending) return result.data;
+  else return undefined;
```

Comment:
maybe it can be simplified in this way (?)

```suggestion
//we are considering, and returning, only processed messages. A processed message has isPending === false
const result = messageMetadataSchema
  .and(z.object({ isPending: z.literal(false) }))
  .catch(undefined)  
  .parse(input)
return result;
```

Reply:
I've tested it but there are types problems with the catch. I've decided to keep the if statement